app/domain/authentication/authn_oidc/authenticate.rb

```diff
@@ -30,17 +31,20 @@ def access_token(input)
             required_variable_names: required_variable_names
         )
 
-        # Prepare ID token introspect request
-
-        # send id token to OIDC Provider
-
-        # Get JSON from OIDC Provider
-
-        # Validate ID Token is active
+        id_token_attributes = @decode_and_verify_id_token.(
+          provider_uri: oidc_secrets["provider-uri"],
+            id_token_jwt: request_body.id_token
+        )
```

Comment:
Regarding the structure of the `call` method here: A couple points about how `CommandClass` is intended to be used.  

1. Currently `call` calls `access_token`, which has the actual high-level logic.  We probably want to move the logic that's currently in `access_token` into `call`.  Remember the name of the command class should be enough to summarize what it does.  If `Authenticate` is not clear enough, the command class itself can be named `FetchAccessToken` or `FetchAuthenticationToken`, etc.  `call` then carries out the steps implicit in that name.

2. The call method itself should read like a table of contents, or a series of sentences.  One of the huge benefits of command classes is that they reveal your high-level logic as clearly as a bulleted list of steps would, in documentation.  The philosphy is: The code itself should be as clear as you describing in English what it does.  For example, here's the `call` method in the [K8s authenticator](https://github.com/cyberark/conjur/blob/6b92244b7481bcf1f72f2ad3896dddcbcf0675f0/app/domain/authentication/authn_k8s/authenticator.rb#L18):

```
      def call
        validate_cert_exists
        validate_the_request
        validate_header_cert
      end
```

In the current implementation, things are starting to get away from us just a bit... we have the details of parameter passing, local variables, etc, obscuring our high-level logic.  The ideal is: You never have to read _code_ to figure out what the code does.  Instead, you're reading English sentences (in the form of method names) to figure it out.  *If* you are interested in more detail, you can jump to those method definitions.